torch/distributed/rendezvous.py

```diff
@@ -51,19 +51,19 @@ def register_rendezvous_handler(scheme, handler):
     _rendezvous_handlers[scheme] = handler
 
 
-def rendezvous(url: str, rank: int = -1, world_size: int = -1, **kwargs):
+def rendezvous(url: str, rank: int = -1, world_size: Optional[int] = None, **kwargs):
```

Comment:
curious, why do we need to convert it to optional? Will it work if we just use the default value -1 for dynamic world size?